src/cpu/cpu.cpp

```diff
@@ -1619,44 +1723,54 @@ void CPU_SET_CRX(Bitu cr, Bitu value)
 			LOG(LOG_CPU, LOG_NORMAL)("Protected mode");
 			PAGING_Enable((value & CR0_PAGING) > 0);
 
-			if (!CPU_AutoDetermineMode.auto_core &&
-			    !CPU_AutoDetermineMode.auto_cycles) {
+			if (!auto_determine_mode.auto_core &&
+			    !auto_determine_mode.auto_cycles) {
 				break;
 			}
 
-			if (CPU_AutoDetermineMode.auto_cycles) {
-				CPU_CycleAutoAdjust = true;
-				CPU_CycleLeft       = 0;
-				CPU_Cycles          = 0;
-				CPU_OldCycleMax     = CPU_CycleMax;
+			if (legacy_cycles_mode) {
+				if (auto_determine_mode.auto_cycles) {
+					CPU_CycleAutoAdjust = true;
+					CPU_CycleLeft       = 0;
+					CPU_Cycles          = 0;
+					old_cycle_max       = CPU_CycleMax;
+
+					GFX_NotifyCyclesChanged();
 
-				GFX_NotifyCyclesChanged();
+					if (!displayed_max_cycles_warning) {
+						LOG_WARNING(
+								"CPU: Switched to max cycles for protected mode program. "
+								"Try setting a fixed cycles value if you're getting audio "
+								"drop-outs or the programs runs too fast.");
```

Comment:
Do we want to also display this warning when using modern cycles setting?

Reply:
Yeah, good idea.